### ISSUE TITLE
server.py: _sync_config() runs after _detect_native_tool_support(), silently disabling native tool format

### ISSUE THREAD

**Issue by reidperyam** [raullenchai/vllm-mlx#225]
### Rapid-MLX version

0.6.5

### Hardware

 Mac Mini M4 Pro, 48 GB

### macOS version

macOS 26.4.1

### Python version

3.12.12

### Model

mlx-community/Qwen3.5-9B-4bit (also reproduces with mlx-community/Qwen3.6-35B-A3B-4bit-DWQ

### Full serve command

rapid-mlx serve mlx-community/Qwen3.5-9B-4bit --port 8001 --enable-auto-tool-choice --tool-call-parser qwen3_coder

### Streaming or non-streaming?

Streaming (`stream: true`)

### What happened? What did you expect?

Actual: After the first assistant tool_calls + tool round-trip, ~40% of subsequent assistant turns emit `[Calling tool: name({"key":"value"})]` as plain content text instead of structured tool_calls. stopReason=stop, no tool_calls field. Multi-agent workloads (50+ tool turns/burst) reliably trigger this within the first few rounds.

Expected: All assistant tool calls emitted as structured tool_calls (delta.tool_calls in streaming, message.tool_calls in non-streaming) regardless of conversation depth.

Root cause: Native tool format is silently disabled at startup. _detect_native_tool_support() at server.py:499 runs BEFORE _sync_config() at server.py:559. When detection reads cfg.enable_auto_tool_choice and cfg.tool_call_parser via get_config(), both are still at their dataclass defaults (False, None) because the globals haven't been copied into the singleton yet. The guard `if not cfg.enable_auto_tool_choice or not cfg.tool_call_parser: return False` short-circuits, _engine.preserve_native_tool_format = False, and the "Native tool format enabled for parser: ..." log never prints.

Downstream effect: api/utils.py::process_messages takes the prose-conversion fallback path (lines 672, 698, 727), serializing assistant tool_calls as `[Calling tool: name(args)]` and tool messages as `[Tool Result (id)]: ...` into the prompt history. Models then see prose-format examples in context and mimic the pattern on subsequent turns. Looks like a model failure but the model is faithfully completing a malformed prompt — exactly what hermes_tool_parser.py:65-68 already documents ("Without this, tool history is converted to '[Calling tool: ...]' text, which causes the model to mimic that text format").

Verification that the parser flag is correct:
  >>> from vllm_mlx.tool_parsers.abstract_tool_parser import ToolParserManager
  >>> p = ToolParserManager.get_tool_parser('qwen3_coder')
  >>> p.__name__, p.SUPPORTS_NATIVE_TOOL_FORMAT
  ('HermesToolParser', True)

The parser correctly reports native support; detection just never gets to ask it.

Fix: move _sync_config() to before _detect_native_tool_support() in load_model() (server.py around line 499). One-line move. Affects all parsers with SUPPORTS_NATIVE_TOOL_FORMAT = True (deepseek, deepseekv31, functionary, granite, harmony, hermes/qwen3_coder, kimi, llama, mistral, qwen3coder_xml, seed_oss).

After patch: startup log shows "Native tool format enabled for parser: qwen3_coder", _engine.preserve_native_tool_format == True, multi-turn tool-using sessions produce 50+ structured tool_calls in a row with zero text leaks. Confirmed on both Qwen3.5-9B-4bit and Qwen3.6-35B-A3B-4bit-DWQ.

Note: --enable-tool-logits-bias does NOT mask the bug. Bias steers token selection but the model still sees prose-format history in context, so leaks at the same rate.

### Minimal reproduction

1. Start server:
   rapid-mlx serve mlx-community/Qwen3.5-9B-4bit \
     --port 8001 \
     --enable-auto-tool-choice \
     --tool-call-parser qwen3_coder

2. Confirm bug active: startup log does NOT contain "Native tool format enabled for parser: qwen3_coder".

3. Send a request that exercises the multi-turn tool-call path. The bug is stochastic per-turn (~40%) but reliably triggered after a few rounds:

   curl -s http://localhost:8001/v1/chat/completions \
     -H "Content-Type: application/json" \
     -d '{
       "model": "mlx-community/Qwen3.5-9B-4bit",
       "messages": [
         {"role":"user","content":"Read /tmp/foo and summarize"},
         {"role":"assistant","content":null,"tool_calls":[{"id":"call_a","type":"function","function":{"name":"read","arguments":"{\"path\":\"/tmp/foo\"}"}}]},
         {"role":"tool","tool_call_id":"call_a","content":"file contents..."},
         {"role":"user","content":"Now read /tmp/bar"}
       ],
       "tools":[{"type":"function","function":{"name":"read","parameters":{"type":"object","properties":{"path":{"type":"string"}},"required":["path"]}}}],
       "stream": true
     }'

4. Some fraction of responses will stream `delta.content: "["` followed by the literal text `[Calling tool: read({"path":"/tmp/bar"})]` instead of `delta.tool_calls`.

5. Patching server.py to call _sync_config() before _detect_native_tool_support() makes the bug disappear entirely (validated across 50+ consecutive multi-turn tool calls).

### Server logs / error output

```shell
INFO:vllm_mlx.model_auto_config:Auto-detected model family 'qwen3' → tool_call_parser=hermes, reasoning_parser=qwen3
INFO:vllm_mlx.server:Loading model with BatchedEngine: mlx-community/Qwen3.5-9B-4bit
INFO:vllm_mlx.server:Model loaded: mlx-community/Qwen3.5-9B-4bit
INFO:vllm_mlx.server:Default max tokens: 8192

# (Note: missing "Native tool format enabled for parser: qwen3_coder" — never printed because _detect_native_tool_support() returned False due to startup ordering)

# Example leaking response stream:
INFO:vllm_mlx.routes.chat:[REQUEST] POST /v1/chat/completions stream=True model='mlx-community/Qwen3.5-9B-4bit' msgs=4 roles=['system', 'user', 'assistant', 'tool'] tools=7
INFO:vllm_mlx.routes.chat:[SSE-ROLE] data: {"choices":[{"index":0,"delta":{"role":"assistant"}}]}
# (no [SSE-TC] line — text content streamed instead)
INFO:vllm_mlx.engine_core:[stream_outputs] xxx finished normally, 29 tokens in 1.3s
# Final assistant content: "[Calling tool: read({\"path\":\"/Users/.../skills/queue-processor/SKILL.md\"})]"

# Patched startup:
INFO:vllm_mlx.server:Native tool format enabled for parser: qwen3_coder
```

**Comment by raullenchai** [comment]
Fixed in [#230](https://github.com/raullenchai/Rapid-MLX/pull/230) (squash-merged as `10b764b`) and shipped in [v0.6.12](https://github.com/raullenchai/Rapid-MLX/releases/tag/v0.6.12) — thanks for the precise root-cause analysis and reproduction, that made this a one-line move + 4 regression tests instead of a debugging hunt.

The fix moves `_sync_config()` to immediately after engine construction, so detection sees the synced cfg. Second `_sync_config()` at end of `load_model` retained as a defensive safety net per round-2 review feedback.

Verification:
- 3 new regression tests (`tests/test_server_load_model_order.py`): contract, integration, idempotency
- pr_validate full pipeline green: `targeted_tests` 68/68 passed, `full_unit` 2148/2148 passed, `stress_e2e_bench` 3×3 matrix passed (qwen3-smoke + Qwen3.5-35B-A3B-8bit + Qwen3.6-27B-4bit)
- GitHub CI all 7 checks SUCCESS

Once `pip install rapid-mlx==0.6.12` (or `brew upgrade rapid-mlx`) is available — should be a few minutes after the release publish — the startup log should show `Native tool format enabled for parser: qwen3_coder` for your serve command, and multi-turn tool sessions should produce structured `tool_calls` instead of `[Calling tool: ...]` text.
